gitbook/graphics-draw.md

```diff
@@ -2,144 +2,130 @@
 
-SFML also provides a way to draw to a texture instead of directly to a window. To do so, use a [RenderTexture]({{book.api}}/RenderTexture.html) instead of a [RenderWindow]({{book.api}}/RenderWindow.html). It has the same functions for drawing, inherited from their common base: [RenderTarget]({{book.api}}/RenderTarget.html). 
+CrSFML also provides a way to draw to a texture instead of directly to a window. To do so, use a [RenderTexture]({{book.api}}/RenderTexture.html) instead of a [RenderWindow]({{book.api}}/RenderWindow.html). It has the same functions for drawing, inherited from their common base: [RenderTarget]({{book.api}}/RenderTarget.html).
 
-```
-// create a 500x500 render-texture
-sf::RenderTexture renderTexture;
-if (!renderTexture.create(500, 500))
-{
-    // error...
-}
-
-// drawing uses the same functions
-renderTexture.clear();
-renderTexture.draw(sprite); // or any other drawable
-renderTexture.display();
-
-// get the target texture (where the stuff has been drawn)
-const sf::Texture& texture = renderTexture.getTexture();
-
-// draw it to the window
-sf::Sprite sprite(texture);
-window.draw(sprite);
-```
+```ruby
+# create a 500x500 render-texture
+render_texture = SF::RenderTexture.new(500, 500)
+unless render_texture
+    # error...
+end
 
-The `getTexture` function returns a read-only texture, which means that you can only use it, not modify it. If you need to modify it before using it, you can copy it to your own [Texture]({{book.api}}/Texture.html) instance and modify that instead. 
+# drawing uses the same functions
+render_texture.clear(SF::Color::Blue)
+render_texture.draw(sprite) # or any other drawable
+render_texture.display
 
-[RenderTexture]({{book.api}}/RenderTexture.html) also has the same functions as [RenderWindow]({{book.api}}/RenderWindow.html) for handling views and OpenGL (see the corresponding tutorials for more details). If you use OpenGL to draw to the render-texture, you can request creation of a depth buffer by using the third optional argument of the `create` function. 
+# get the target texture (where the stuff has been drawn)
+texture = render_texture.texture
 
+# draw it to the window
+sprite = SF::Sprite.new(texture)
+window.draw(sprite)
 ```
-renderTexture.create(500, 500, true); // enable depth buffer
+
+The `texture` function returns a read-only texture, which means that you can only use it, not modify it. If you need to modify it before using it, you can copy it to your own [Texture]({{book.api}}/Texture.html) instance and modify that instead.
+
+[RenderTexture]({{book.api}}/RenderTexture.html) also has the same functions as [RenderWindow]({{book.api}}/RenderWindow.html) for handling views and OpenGL (see the corresponding tutorials for more details). If you use OpenGL to draw to the render-texture, you can request creation of a depth buffer by using the third optional argument of the constructor function.
+
+```ruby
+SF::RenderTexture.new(500, 500, true) # enable depth buffer
```

Comment:
There is no third argument in `SF::RenderTexture.new(500, 500, true)` to use OpenGL. Actually there is no `create` method, but instantiating a class with `new` works almost the same.